### DISCUSSION TITLE
Regex Generator Idea

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#68075] [Ideas] [answered]
@stephentoub I had an idea this morning that I wanted to pass your way.  Maybe it is irrelevant but I wanted to at least toss it in the bucket for consideration.

I seem to recall you mentioning once that regexs like ```ab``` and ```(a)(b)``` are technically different regexes because of how grouping will be handled between them.

When I use regexes, I use parens so that I am very clear about how I'm capturing things, but I never index into the groups of a captured regex using their numeric index.  I either want the whole value of text that the pattern matched or I want to extract explicit named captures.  I speculate that other developers use regexes in a similar way as well.

If ```ab``` is more optimizable than ```(a)(b)```, it would be great to have a flag like ```IgnorePatternGroups``` (similar to ```IgnorePatternWhitespace```) that can be used to tell the regex engine that I don't care about how the groups are returned as long as my named captures still have their values.

**Comment by stephentoub** [comment, marked as answer]
That already exists:
https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regexoptions?view=net-6.0#system-text-regularexpressions-regexoptions-explicitcapture

**Reply by TonyValenti** [reply]
Oh wow!  I didn't realize that!

**Comment by khaimran** [comment]
WOW Its an Excellent!!!
Our [Best Interior Designers in Noida](https://thegoodwood.in) not solely design homes, but additionally workplaces, schools, churches, and hospitals. If you’re looking to make your house a pleasurable place to be in or you’re looking for a transformation of a space in your office or building at work, call today to get an appointment with a specialist.
